misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/RangeFormatter.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.formatter.core;
+
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocuments;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The abstract class that is extended by all range formatting test classes.
+ *
+ * @since 1.2.10
+ */
+public abstract class RangeFormatter {
+    private final Path resourceDirectory = Paths.get("src").resolve("test").resolve("resources").toAbsolutePath();
+    private Path buildDirectory = Paths.get("build").toAbsolutePath().normalize();
+    private static final String ASSERT_DIR = "assert";
+    private static final String SOURCE_DIR = "source";
+
+    @Test(dataProvider = "test-file-provider")
+    public void test(String source, int[][] positions)
+            throws IOException, FormatterException {
+        Path assertFilePath = Paths.get(resourceDirectory.toString(), this.getTestResourceDir(), ASSERT_DIR, source);
+        Path sourceFilePath = Paths.get(resourceDirectory.toString(), this.getTestResourceDir(), SOURCE_DIR, source);
+
+        String content = getSourceText(sourceFilePath);
+        TextDocument textDocument = TextDocuments.from(content);
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument, sourceFilePath.toString());
+        for (int[] position : positions) {
+            LinePosition startPos = LinePosition.from(position[0], position[1]);
+            LinePosition endPos = LinePosition.from(position[2], position[3]);
+            LineRange lineRange = LineRange.from(null, startPos, endPos);
+            syntaxTree = Formatter.format(syntaxTree, lineRange);
+        }
+        Assert.assertEquals(syntaxTree.toSourceCode(), getSourceText(assertFilePath));
+    }
+
+    /**
+     * Test the formatting functionality for parser test cases.
+     *
+     * @param sourcePath Source path of the parser test
+     */
+    public void testParserResources(String sourcePath) throws IOException, FormatterException {
+        Path filePath = Paths.get(sourcePath);
+        String content = getSourceText(filePath);
+        TextDocument textDocument = TextDocuments.from(content);
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument);
+        if (!syntaxTree.hasDiagnostics()) {
```

Comment:
this will make the tests to get skipped silently, if someone introduces changes to the file.
Shall we remove this check?